### DISCUSSION TITLE
Experiencing slower running speed using HGTConv from v2.5.0 compared to v2.3.0

### DISCUSSION THREAD

**Discussion by junhongmit** [pyg-team/pytorch_geometric#8954] [General]
Hi, I noticed HGTConv in [v2.5.0](https://pytorch-geometric.readthedocs.io/en/2.5.0/_modules/torch_geometric/nn/conv/hgt_conv.html) has been improved over the previous version, [v2.3.0](https://pytorch-geometric.readthedocs.io/en/2.3.0/_modules/torch_geometric/nn/conv/hgt_conv.html), with notably changes like using HeteroDictLinear and slicing node features into single tensor. However, I am experiencing roughly 4 times slower training speed (20 secs -> 80 secs) and 2 times slower inference speed (37 secs -> 66 secs) when simply replacing the HGTConv with the new convolution operator in v2.5.0. See the following speed comparison. I am pretty sure this is originated from HGTConv itself only. Because, after I replace the newer hgt_conv.py with the old hgt_conv.py from v2.3.0, the speed becomes normal again. Is it because the tensor slicing costs more time in the newer version?
![image](https://github.com/pyg-team/pytorch_geometric/assets/35260835/d69abec1-5fd2-4682-8711-8bdb033d81c1)
![image](https://github.com/pyg-team/pytorch_geometric/assets/35260835/6da788b9-cac0-43cc-b345-f02b8d8c0bb6)

**Comment by rusty1s** [comment]
Do you mind sharing a small example that reproduces this? Happy to take a look.
